ouroboros-consensus-shelley/test/Test/Consensus/Shelley/Ledger/Golden.hs

```diff
@@ -876,18 +1077,546 @@ test_golden_LedgerState = goldenTestCBOR
+    , TkBytes "\195\232\167\212"
+    , TkInt 40
+    , TkListLen 1
+    , TkBytes "\246\216G\149"
+    , TkInt 42
+    , TkListLen 1
+    , TkBytes "\SO\152\236\221"
+    , TkInt 44
+    , TkListLen 1
+    , TkBytes "!Mlu"
+    , TkInt 46
+    , TkListLen 1
+    , TkBytes "Sk\188V"
+    , TkInt 48
+    , TkListLen 1
+    , TkBytes "r\198Y\SO"
+    , TkInt 50
+    , TkListLen 1
+    , TkBytes "\135\151\&0\144"
+    , TkInt 52
+    , TkListLen 1
+    , TkBytes "\195\232\167\212"
+    , TkInt 54
+    , TkListLen 1
+    , TkBytes "\246\216G\149"
+    , TkInt 56
+    , TkListLen 1
+    , TkBytes "\SO\152\236\221"
+    , TkInt 58
+    , TkListLen 1
+    , TkBytes "!Mlu"
+    , TkInt 60
+    , TkListLen 1
+    , TkBytes "Sk\188V"
+    , TkInt 62
+    , TkListLen 1
+    , TkBytes "r\198Y\SO"
+    , TkInt 64
+    , TkListLen 1
+    , TkBytes "\135\151\&0\144"
+    , TkInt 66
+    , TkListLen 1
+    , TkBytes "\195\232\167\212"
+    , TkInt 68
+    , TkListLen 1
+    , TkBytes "\246\216G\149"
+    , TkInt 70
+    , TkListLen 1
+    , TkBytes "\SO\152\236\221"
+    , TkInt 72
+    , TkListLen 1
+    , TkBytes "!Mlu"
+    , TkInt 74
+    , TkListLen 1
+    , TkBytes "Sk\188V"
+    , TkInt 76
+    , TkListLen 1
+    , TkBytes "r\198Y\SO"
+    , TkInt 78
+    , TkListLen 1
+    , TkBytes "\135\151\&0\144"
+    , TkInt 80
+    , TkListLen 1
+    , TkBytes "\195\232\167\212"
+    , TkInt 82
+    , TkListLen 1
+    , TkBytes "\246\216G\149"
+    , TkInt 84
+    , TkListLen 1
+    , TkBytes "\SO\152\236\221"
+    , TkInt 86
+    , TkListLen 1
+    , TkBytes "!Mlu"
+    , TkInt 88
+    , TkListLen 1
+    , TkBytes "Sk\188V"
+    , TkInt 90
+    , TkListLen 1
+    , TkBytes "r\198Y\SO"
+    , TkInt 92
+    , TkListLen 1
+    , TkBytes "\135\151\&0\144"
+    , TkInt 94
+    , TkListLen 1
+    , TkBytes "\195\232\167\212"
+    , TkInt 96
+    , TkListLen 1
+    , TkBytes "\246\216G\149"
+    , TkInt 98
+    , TkListLen 1
+    , TkBytes "\SO\152\236\221"
+    , TkListLen 3
+    , TkListLen 2
+    , TkInt 0
+    , TkListLen 2
+    , TkListLen 2
+    , TkInt 1
+    , TkInt 1
+    , TkMapLen 1
+    , TkListLen 2
+    , TkInt 1
+    , TkInt 1
+    , TkListLen 6
+    , TkMapLen 1
+    , TkBytes "U\165@\b"
+    , TkInt 1
+    , TkListLen 1
+    , TkListLen 3
+    , TkInt 0
+    , TkInt 1
+    , TkBytes "U\165@\b"
+    , TkListLen 1
+    , TkInt 0
+    , TkListLen 2
+    , TkInt 1
+    , TkBytes "K\245\DC2/4ET\197;\222.\187\140\210\183\227\209`\n\214\&1\195\133\165\215\204\226<w\133E\154"
+    , TkListLen 1
+    , TkInt 0
+    , TkListLen 2
+    , TkInt 1
+    , TkBytes "\219\193\180\201\NUL\255\228\141W[]\165\198\&8\EOT\SOH%\246]\176\254>$IKv\234\152dW\217\134"
+    , TkListLen 0
+    , TkListLen 0
+    ]
+  where
+    exampleExtLedgerState = ExtLedgerState
+      { ledgerState = exampleLedgerState
+      , headerState = exampleHeaderState
+      }
+
+test_golden_Query :: Query Block result -> FlatTerm -> Assertion
+test_golden_Query = goldenTestCBOR encodeShelleyQuery
```

Comment:
Can this become a local helper of `testQueries`? It living here gives the false impression that it is a proper test case while it still takes an argument.